library/aria.c

```diff
@@ -112,8 +112,12 @@ static inline uint32_t aria_p1( uint32_t x )
 #endif /* arm */
 #if defined(__GNUC__) && \
     defined(__i386__) || defined(__amd64__) || defined( __x86_64__)
-/* I couldn't find an Intel equivalent of rev16, so two instructions */
-#define ARIA_P1(x) ARIA_P2( ARIA_P3( x ) )
+static inline uint32_t aria_p1( uint32_t x )
+{
+    uint32_t r;
+    __asm( "xchgb %b0, %h0" : "=q"(r) : "0"(x) );
```

Comment:
I'm only moderately familiar with x86 assembly, and it is unclear to me what's going on here, so I think a comment would be helpful.